### PR TITLE
[Sprint 4] Comments Functionality + Like/Dislike Revisions

### PR DESCRIPTION
**Pull request by maxbalves**
## Description
**Like/Dislike Revisions:**
- Refactored code in order to reuse previously written functions

**Comments Functionality:**
- Users are able to see comments under a post
- Users are able to add comments to a post
- Users have the ability to hide their username or profile picture on their comments
- If a user tries to comment on an invalid or deleted post, no action is done
    - The user is simply taken back to Details page, which then automatically displays warning and etc... 
- Infinite scrolling on comments allows users to scroll down and see more posts.

## Milestones
Milestone/Sprint 3, [Add comments functionality to posts #23](https://github.com/maxbalves/PostBy/issues/23)

## Test Plan
Tested by:
- Creating a post and adding comments through many accounts
- At the end, checked if deleting a post would also delete comments

**Demo:**

## Future Plans
- My next goal is to revise the account deletion feature in order to make sure comments, likes, dislikes, and posts, all together are being deleted properly.
- Future optional goals:
    - Improve UI of comments by implementing UIScrollView in Details Page
    - Revise the code in the Like/Dislike Safety feature in order to improve responsiveness
    - etc...

### CODE REVIEW

File: PostBy/View Controllers/DetailsViewController.m [hunk 1]
```diff
@@ -42,8 +53,19 @@ - (void)viewDidLoad {
...
     self.EDIT_SEGUE = @"EditPostSegue";
     self.MAP_SEGUE = @"DetailsShowMap";
+    self.COMMENT_SEGUE = @"CommentPostSegue";
     
     [self setUpUI];
+    
+    self.commentsTableView.dataSource = self;
+    self.commentsTableView.delegate = self;
+    
+    self.MAX_COMMENTS_SHOWN = 8;
```
Comment (raozhang-fb) [inline PostBy/View Controllers/DetailsViewController.m:63]: 👍

File: PostBy/Models/Comment.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Comment *newComment = [Comment new];
+    newComment.post = post;
+    newComment.author = [PFUser currentUser];
+    newComment.commentText = text;
+    newComment.hideUsername = hideUsername;
+    newComment.hideProfilePic = hideProfilePic;
+    
+    [newComment saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        // Add comment to its post's "comments" relation
+        PFRelation *postCommentsRelation = [post relationForKey:@"comments"];
```
Comment (raozhang-fb) [inline PostBy/Models/Comment.m:48]: A kindly reminder -- If the "comments" is only referenced in this function, it is fine to keep it. However, it is better to set up a global variable for it, if it is also used in other functions or across files.

**Review by raozhang-fb — COMMENTED** [review]
Pretty clear PR with comments functionality added and like/dislike updated.
Good job! 👍

File: PostBy/Models/Comment.m (hunk 2 above)
Comment (maxbalves) [reply PostBy/Models/Comment.m:48]: Good point Rao! A lot of the relation names are referenced across file. I will create an issue/goal so I can later revise all of them at once and create their respective global variables.

**Pull request merged**